source/server/server.cc

```diff
@@ -148,14 +148,22 @@ void InstanceImpl::flushStats() {
   // A shutdown initiated before this callback may prevent this from being called as per
   // the semantics documented in ThreadLocal's runOnAllThreads method.
   stats_store_.mergeHistograms([this]() -> void {
-    HotRestart::GetParentStatsInfo info;
-    restarter_.getParentStats(info);
+    uint64_t parent_memory_allocated = 0;
+    uint64_t parent_connections = 0;
+    std::unique_ptr<envoy::HotRestartMessage> wrapper_msg = restarter_.getParentStats();
+    // getParentStats() will happily and cleanly return nullptr if we have no parent.
+    if (wrapper_msg) {
+      restarter_.mergeParentStats(stats_store_, wrapper_msg->reply().stats());
```

Comment:
This is where I would just call a static utility function instead of having an interface/object, etc.

Reply:
(answered in the other comment)